### PR TITLE
Refactor methods from MachineShopSimulator to Machine

### LINKED ISSUE #4: Refactor comments in MachineShopSimulator Class

### LINKED ISSUE #5: Machine is a Data class
Machine is a classes that primarily just have get and set methods, and could have more existing methods that use them pulled into them. Through refactoring our thoughts are that more and more code will be moved into these classes from other places. This  will make these classes actually do more instead of just holding a few things.

### PR DESCRIPTION
**Pull request by burde041**
## Description
Machine seems to primarily be a Data class, so it can have some methods from MachineShopSimulator, which coincidentally is a big class.

## Related Issue
This PR is addressing issues #5 and partially issue #1 and issue #4

## Motivation and Context
This change is required to make Machine so it is not just a Data class.

This addresses the smell of having a class be too long, be a data class, and have unnecessary comments.

The issue was just about Machine being a Data class.

## Smell impact

This eliminates the smell of Machine being a Data class and reduces the smell of MachineShopSimulator being too big and having inconsistent comments.

## Testing & Screenshots:

coverage

- [x] All new and existing tests passed.

### CODE REVIEW

**Review by smit9025 — APPROVED** [review]
Commit messages are clear. I think what has been done with extracting methods from machineshopsimulator.java to machine.java has been done properly and reduces the data class smell and helps make machineshopsimulator.java smaller. I can't think of any suggestions to make.

File: src/applications/Machine.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package applications;
 
 import dataStructures.LinkedQueue;
+import jdk.jfr.Event;
```
Comment (NicMcPhee) [inline src/applications/Machine.java]: I bet this is an autocomplete thing, and that you almost certainly don't want it.

File: src/applications/MachineShopSimulator.java [hunk 2]
```diff
@@ -21,171 +20,115 @@
...
+        else {
             int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
-            // put on machine p's wait queue
             machine[p].getJobQ().put(theJob);
             theJob.setArrivalTime(timeNow);
-            // if p idle, schedule immediately
-            if (eList.nextEventTime(p) == largeTime) {// machine is idle
-                changeState(p);
+            if (eList.nextEventTime(p) == largeTime) {
+                Machine.changeState(p, machine[p], eList, timeNow);
```
Comment (NicMcPhee) [inline src/applications/MachineShopSimulator.java]: Would it make more sense to make `changeState` a non-static method called on the desired machine?

```suggestion
                machine[p].changeState(p, eList, timeNow);
```

File: src/applications/MachineShopSimulator.java [hunk 3]
```diff
@@ -21,171 +20,115 @@
...
-
-        // Move this to startShop when ready
         setMachineChangeOverTimes(specification);
-
-        // Move this to startShop when ready
         setUpJobs(specification);
 
         for (int p = 1; p <= numMachines; p++)
-            changeState(p);
+            Machine.changeState(p, machine[p], eList, timeNow);
```
Comment (NicMcPhee) [inline src/applications/MachineShopSimulator.java]: A similar comment would apply here:

```suggestion
            machine[p].changeState(p, eList, timeNow);
```

File: src/applications/MachineShopSimulator.java [hunk 4]
```diff
@@ -21,171 +20,115 @@
...
-        simulationResults.setTotalWaitTimePerMachine(totalWaitTimePerMachine);
-    }
-
-    private void setNumTasksPerMachine(SimulationResults simulationResults) {
-        int[] numTasksPerMachine = new int[numMachines+1];
-        for (int i=1; i<=numMachines; ++i) {
-            numTasksPerMachine[i] = machine[i].getNumTasks();
-        }
-        simulationResults.setNumTasksPerMachine(numTasksPerMachine);
+        Machine.setTotalAndNumTasksPerMachine(simulationResults, new int[numMachines + 1], machine);
```
Comment (NicMcPhee) [inline src/applications/MachineShopSimulator.java:112]: This could also be made a method on `machine` instead of a static method.

**Review by NicMcPhee — CHANGES_REQUESTED** [review]
Nice work!

In general you want to avoid `static` methods if you can help it – they're not very "object-oriented" and definitely aren't "best practice" in the OO universe. Sometimes you _need_ them, but you tend to want to consider them a smell and try to reduce rather than increase their numbers. In this case you bring in some static methods that definitely don't _need_ to be static, so you probably want to avoid making them static.

**Pull request merged**